mne/stats/regression.py

```diff
@@ -134,3 +136,193 @@ def _fit_lm(data, design_matrix, names):
+    potentials and allowing for continuous/scalar predictors. Internally, this
+    constructs a predictor matrix X of size n_samples * (n_conds * window
+    length), solving the linear system Y = bX and returning b as evoked-like
+    time series split by condition.
+
+    See Smith, N. J., & Kutas, M. (2015). Regression-based estimation of ERP
+    waveforms: II. Non-linear effects, overlap correction, and practical
+    considerations. Psychophysiology, 52(2), 169-189.
+
+    Parameters
+    ----------
+    raw : instance of Raw
+        A raw object. Warning: be very careful about data that is not
+        downsampled, as the resulting matrices can be enormous and easily
+        overload your computer. Typically, 100 hz sampling rate is
+        appropriate - or using the decim keyword (see below).
+    events : np.ndarray of int, shape(n_events, 3)
+        An array where the first column corresponds to samples in raw
+        and the last to integer codes in event_id.
+    event_id : dict
+        As in Epochs; a dictionary where the values may be integers or
+        list-like collections of integers, corresponding to the 3rd column of
+        events, and the keys are condition names.
+    tmin : float | dict
+        If float, gives the lower limit (in seconds) for the time window for
+        which all event types' effects are estimated. If a dict, can be used to
+        specify time windows for specific event types: keys correspond to keys
+        in event_id and/or covariates; for missing values, the default (-.1) is
+        used.
+    tmax : float | dict
+        If float, gives the upper limit (in seconds) for the time window for
+        which all event types' effects are estimated. If a dict, can be used to
+        specify time windows for specific event types: keys correspond to keys
+        in event_id and/or covariates; for missing values, the default (1.) is
+        used.
+    covariates : dict-like | None
+        If dict-like, values have to be array-like and of the same length as
+        the columns in ```events```. Keys correspond to additional event
+        types/conditions to be estimated and are matched with the time points
+        given by the first column of ```events```. If None, only binary events
+        (from event_id) are used.
+    reject : None | dict
+        Activate rejection parameters based on peak-to-peak amplitude in
+        continuously selected subwindows. If reject is None, no rejection is
+        done. If dict, keys are types ('grad' | 'mag' | 'eeg' | 'eog' | 'ecg')
+        and values are the maximal peak-to-peak values to select rejected
+        epochs. E.g. reject = dict(grad=4000e-12, # T / m (gradiometers)
+                                   mag=4e-11, # T (magnetometers)
+                                   eeg=40e-5, # uV (EEG channels)
+                                   eog=250e-5 # uV (EOG channels))
+    tstep : float
+        Length of windows for peak-to-peak detection.
```

Comment:
I guess that's a separate issue but when would you use such an estimator?

On Tue, Jul 21, 2015 at 5:30 PM, jona-sassenhagen <notifications@github.com>
wrote:

> In mne/stats/regression.py
> <https://github.com/mne-tools/mne-python/pull/2304#discussion_r35115927>:
>
> > +        If dict-like, values have to be array-like and of the same length as
> > +        the columns in ```events```. Keys correspond to additional event
> > +        types/conditions to be estimated and are matched with the time points
> > +        given by the first column of ```events```. If None, only binary events
> > +        (from event_id) are used.
> > +    reject : None | dict
> > +        Activate rejection parameters based on peak-to-peak amplitude in
> > +        continuously selected subwindows. If reject is None, no rejection is
> > +        done. If dict, keys are types ('grad' | 'mag' | 'eeg' | 'eog' | 'ecg')
> > +        and values are the maximal peak-to-peak values to select rejected
> > +        epochs. E.g. reject = dict(grad=4000e-12, # T / m (gradiometers)
> > +                                   mag=4e-11, # T (magnetometers)
> > +                                   eeg=40e-5, # uV (EEG channels)
> > +                                   eog=250e-5 # uV (EOG channels))
> > +    tstep : float
> > +        Length of windows for peak-to-peak detection.
>
> Aside: maybe it would be cool to allow other estimators than peak-to-peak
> for this helper, such as max(abs(?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mne-tools/mne-python/pull/2304/files#r35115927>.
>